magento/magento2
# Fixed the wrong behavior for a prompt modal when a user clicks on the modal overlay

### Description (*)
The prompt modal performs the confirm action when a user clicks on the modal overlay.
This PR fixes this issue and adds the `false` argument for `closeModal()` method when a user clicks on the modal overlay.

### Related Pull Requests

The issue starts reproducing after the changes from this https://github.com/magento/magento2/commit/6167b0cf306689cae3f2309e65ab9653e1e69252 commit.

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#1002: Confirm Save Preview is performed if click outside of the Search Adobe Stock when File Name window is opened

### Manual testing scenarios (*)
The manual testing scenarios and how to reproduce this issue is described in this issue magento/adobe-stock-integration#1002.

Also, it could be checked quickly with the following JS code via a browser console.

```javascript
require(['jquery', 'Magento_Ui/js/modal/prompt'], function ($, prompt) {
    prompt({
        title: 'Prompt Title',
        content: $('.prompt-modal-content'),
        modalClass: 'prompt',
        value: 'Value by default',
        validation: true,
        promptField: '[data-role="promptField"]',
        attributesForm: {
            novalidate: 'novalidate',
            action: ''
        },
        actions: {
            always: function() {
                // do something when the modal is closed
            },
            confirm: function () {
                alert('Confirmation');
            },
            cancel: function () {
                // do something when the cancel button is clicked
            }
        }
    });

    jQuery('.prompt + .modals-overlay').click()
})
```
This code creates the Prompt popup and clicks on overlay.
In 2.4-develop you will see the alert with "Confirmation" that is triggered by the confirm action.

This issue is fixed in this PR and the prompt popup will be closed after clicking on the modal overlay.

### Contribution checklist (*)
